File: cses/leetcode/go/190.go
func reverseBits(num uint32) uint32 {
	var totalBits := 32
	var ans uint32 := 0
	for i := 0; i < totalBits; i++ {
		ans = (ans<<1) | (num&1)
		num >>= 1
	}
	return ans
}

File: cses/leetcode/cpp/190.cpp

#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    uint32_t reverseBits(uint32_t n)
    {
        int totalBits = 32;
        uint32_t ans = 0;
        while (totalBits)
        {
            ans = (ans << 1) | (n & 1);
            n >>= 1; // right-shift to process next bit
            totalBits--;
        }
        return ans;
    }
};
int main()
{
}